app/main/login.py

```diff
@@ -1,20 +1,52 @@
+from __future__ import absolute_import
+
 from . import main
-from flask import render_template, redirect, url_for
+from flask import render_template, redirect, url_for, flash
+from flask_login import logout_user, login_required, login_user, current_user
+from .forms.login_form import LoginForm
+from .. import api_client
 
 
 @main.route('/login', methods=["GET"])
 def render_login():
     template_data = main.config['BASE_TEMPLATE_DATA']
-    return render_template("auth/login.html", **template_data), 200
+    return render_template(
+        "auth/login.html",
+        form=LoginForm(),
+        **template_data), 200
 
 
 @main.route('/login', methods=["POST"])
 def process_login():
-    return redirect(url_for('.dashboard'))
+    form = LoginForm()
+    template_data = main.config['BASE_TEMPLATE_DATA']
+    if form.validate_on_submit():
+
+        user = api_client.users_auth(
+            form.email_address.data,
+            form.password.data)
+
+        if not user:
+            flash(
```

Comment:
We can add categories to `flash` to distinguish between success and error messages. e.g.: 
```
flash('Sorry, we couldn't find a user with that username and password.', 'error')
```
and then in the template use:
```
{% with messages = get_flashed_messages(with_categories=true) %}
```